File: oxva/createData.java
import java.util.HashMap;

public class IneryWriter {
    public static void main(String[] args) {
        // setup JSON RPC
        String jsonRpc = "http://localhost:8888";

        // setup private key
        String privateKey = "your_private_key";

        // setup API
        HashMap<String, Object> apiMap = new HashMap<>();
        apiMap.put("rpc", jsonRpc);
        apiMap.put("signatureProvider", privateKey);

        // the account and actor
        String account = "your_account_name";
        String actor = "your_actor_name";

        // data for the function call
        HashMap<String, Object> dataMap = new HashMap<>();
        dataMap.put("id", 1);
        dataMap.put("user", actor);
        dataMap.put("value", "Data from ineryjs");

        // make the function call
        HashMap<String, Object> actionMap = new HashMap<>();
        actionMap.put("account", account);
        actionMap.put("name", "create");
        actionMap.put("authorization", new HashMap<String, String>() {{ 
            put("actor", actor);
            put("permission", "active");
        }});
        actionMap.put("data", dataMap);

        HashMap<String, Object> transactMap = new HashMap<>();
        transactMap.put("actions", new HashMap<String, Object>[]{actionMap});

        HashMap<String, Object> optionsMap = new HashMap<>();
        optionsMap.put("broadcast", true);
        optionsMap.put("sign", true);

        apiMap.put("transact", transactMap);
        apiMap.put("options", optionsMap);

        // result
        String result;
        try {
            result = apiMap.get("transact").toString();
        } catch (Exception e) {
            System.out.println("Failed to create data: " + e);
            return;
        }

        System.out.println("Transaction: " + result);
    }
}

File: alphadigger/create_data.py
import os
import json
from dotenv import load_dotenv
from api_client.cline import Cline
from api_client import keys
from api_client.utils import *

os.environ["NODE_URL"] = "http://00.000.00.000:8888"
os.environ["PRIVATE_KEY"] = "xxxxx"
os.environ["INERY_ACCOUNT"] = "alphadigger"

rpc_url = os.environ.get("NODE_URL")

json_rpc = JsonRpc(rpc_url)
private_key = os.environ.get("PRIVATE_KEY")
account_name = os.environ.get("INERY_ACCOUNT")
actor_name = os.environ.get("INERY_ACCOUNT")
signature = JsSignatureProvider([private_key])

api_client = ApiClient(rpc=json_rpc, signature_provider=signature)

async def create_data(id, user, data):
    try:
        transaction = await api_client.transact(actions=[{
            "account": account_name,
            "name": "create",
            "authorization": [{
                "actor": actor_name,
                "permission": "active"
            }],
            "data": {
                "id": id,
                "user": user,
                "data": data
            }
        }], broadcast=True, sign=True)

        print(transaction)
        print(transaction["processed"]["action_traces"][0]["console"])
    except RpcError as e:
        print(e)

create_data(10, account_name, "Create new Data")
